lib/OrgFileProcessor.js

```diff
@@ -0,0 +1,240 @@
+          groupedServices = FileProcessorHelper.groupEntitiesPerOperationType(existingServices, configServices);
+          next();
+        });
+      },
+      (next) => {
+        this._createServices({ services: groupedServices[OperationType.CREATE], orgId: fetchedOrg.id }, next);
+      },
+      (next) => {
+        this._updateServices({ services: groupedServices[OperationType.UPDATE], existingServices, orgId: fetchedOrg.id }, next);
+      },
+      (next) => {
+        this._modifyApps({ configApps: configOrg.applications, orgId: fetchedOrg.id }, next);
+      }
+    ], (err) => {
+      if (err) {
+        return done(err);
+      }
+
+      done(null, { id: fetchedOrg.id });
+    });
+  }
+
+  /**
+   * Creates services.
+   * @param {Object} options
+   * @param {Object} options.services Services to be created in config format.
+   * @param {String} options.orgId Org ID.
+   * @param done
+   * @private
+   */
+  _createServices(options, done) {
+    if (isNullOrUndefined(options.services) || isEmpty(options.services)) {
+      return setImmediate(done);
+    }
+
+    const servicesNames = Object.keys(options.services);
+    async.eachSeries(
+      servicesNames,
+      (currName, next) => {
+        this.cliManager.log(LogLevel.INFO, `Creating service: ${currName}`);
+        this.serviceFileProcessor.process(
+          {
+            operation: OperationType.CREATE,
+            parsedData: options.services[currName],
+            name: currName,
+            domainId: options.orgId,
+            domainType: DomainTypes.ORG
+          },
+          next
+        );
+      },
+      done
+    );
+  }
+
+  _updateServices(options, done) {
+    const configServices = options.services;
+    if (isEmpty(configServices)) {
+      return setImmediate(done);
+    }
+
+    const existingServices = options.existingServices;
+    const operation = OperationType.UPDATE;
+
+    async.eachSeries(
+      configServices,
+      (currService, next) => {
+        const nameIdentifier = Object.keys(currService)[0];
+        const existingService = existingServices.find(x => x.name === nameIdentifier);
+        const updateData = currService[nameIdentifier];
+        updateData.name = updateData.name || nameIdentifier;
+        const processOptions = {
+          operation,
+          parsedData: updateData,
+          serviceId: existingService.id,
+          domainId: options.orgId,
+          domainType: DomainTypes.ORG
+        };
+        this.cliManager.log(LogLevel.INFO, `Updating service: ${nameIdentifier}`);
+        this.serviceFileProcessor.process(processOptions, next);
+      },
+      done
+    );
+  }
+
+  _createApps({ apps, orgId }, done) {
+    if (isNullOrUndefined(apps) || isEmpty(apps)) {
+      return setImmediate(done);
+    }
+
+    const appNames = Object.keys(apps);
+    async.eachSeries(
+      appNames,
+      (currName, next) => {
+        this.cliManager.log(LogLevel.INFO, `Creating app: ${currName}`);
```

Comment:
Similar INFO message seems to be missing in the _updateApps method.

Reply:
There are some failing integration tests, though.

Reply:
Yeah, this one turned out to be redundant, actually - the same info was being logged twice.